platform/ios/src/MGLAPIClient.m

```diff
@@ -87,7 +87,7 @@ - (NSURLRequest *)requestForEvents:(NS_ARRAY_OF(MGLMapboxEventAttributes *) *)ev
 }
 
 - (void)setupBaseURL {
-    NSString *testServerURL = [[NSBundle mainBundle] objectForInfoDictionaryKey:@"MGLMetricsTestServerURL"];
+    NSString *testServerURL = [[NSUserDefaults standardUserDefaults] stringForKey:@"MGLMapboxMetricsTestServerURL"];
```

Comment:
Is [this diff](https://gist.github.com/friedbunny/cbc10c75da27dbda4dce5c69df08f388) what you had in mind?

- We have to `registerDefaults` separately in MGLAPIClient because `MGLAPIClientBaseURL` is defined there, not in MGLMapboxEvents.
- NSURL has to be converted to NSData before it can be registered. Is there a better way to do this than NSKeyedArchiver?
- We still have to check if we’re using a non-standard telemetry server (for the certificate checks later), so now there’s a string comparison between the URL and the production URL.